commands/bootstrap.js

```diff
@@ -55,13 +60,58 @@ function api({
 	allsamples = false,
 	link = true,
 	loglevel = 'error',
+	override,
 	verbose = false
 } = {}) {
 	const pkg = packageRoot(cwd);
 
 	if (verbose) loglevel = 'verbose';
 
 	return Promise.resolve()
+		.then(() => {
+			// Override dependencies in package.json, package-lock.json and npm-shrinkwrap.json
+			if (override) {
+				console.log('Overrides with local dependencies from', override);
+				// Collect list of all valid local package tgz archives found
+				const scoped = s => fs.readdirSync(path.join(override, s)).map(d => path.join(s, d));
+				const local = fs
+					.readdirSync(override)
+					.reduce((a, curr) => a.concat(curr.startsWith('@') ? scoped(curr) : curr), [])
+					.filter(d => fs.existsSync(path.join(override, d, 'package.tgz')));
+				if (path.isAbsolute(override)) override = path.relative(cwd, override);
+				['package.json', 'package-lock.json', 'npm-shrinkwrap.json']
+					.map(f => path.join(cwd, f))
+					.filter(f => fs.existsSync(f))
+					.forEach((f, i) => {
+						const lockfile = i > 0;
+						// Restore any detected backups
+						if (fs.existsSync(f + '.bak')) {
+							fs.unlinkSync(f);
+							fs.renameSync(f + '.bak', f);
+						}
```

Comment:
Is this a good idea?  I can see an argument either way.  If the user updates by modifying or pulling fresh copies down, the `.bak` file may be outdated.  However, backing up an already modified version of the file wouldn't be terribly useful, either.  In the second case, however, it should be modifying the exact same files again...  Dunno.

Reply:
I think it's fine personally. We've seen NPM modify shrinkwrap files on their own in the past to the point running override on shrinkwrap a second time after an install has had issues. Not sure if issues remain (it was a while ago) but the backup approach seems safest. Plus this is in the same style as the the old webOS build script and the override option is considered an internal/private option so its usage is entirely under our control.